lua/lsp-selection-range/tests/specs/init_spec.lua

```diff
@@ -13,10 +13,11 @@ end
 
 describe('update_capabilities()', function()
   it('creates new capabilities including "textDocument/selectionRange" and returns them', function()
-    local expectedCapabilities = vim.lsp.protocol.make_client_capabilities()
+    local capabilities = vim.lsp.protocol.make_client_capabilities()
+    local expectedCapabilities = vim.deepcopy(capabilities)
     expectedCapabilities.textDocument.selectionRange = { dynamicRegistration = false }
 
-    local capabilities = lsp_selection_range.update_capabilities(nil)
+    capabilities = lsp_selection_range.update_capabilities(capabilities)
 
     assert.same(expectedCapabilities, capabilities)
   end)
```

Comment:
To cover everything we need to test that:
* when providing with `nil` we
  * return the capabilities from `vim.lsp.protocol.make_client_capabilities` and add the `textDocument.selectionRange.dynamicRegistration = false`
  * raise the deprecation
* when providing with existing capabilities not containing `textDocument.selectionRange.dunamycRegistration` we add it with the value `false`
* when providing with existing capabilities containing `textDocument.selectionRange.dunamycRegistration` we don't change the value